### PR TITLE
Paniukhin HW05

### PR DESCRIPTION
**Pull request by npanuhin**
I deeply apologize for submitting a day after the deadline. I will try to ensure this does not happen in the future 🙏

### CODE REVIEW

File: monads/test/Spec.hs [hunk 1]
```diff
@@ -1,2 +1,70 @@
...
+tests :: TestTree
+tests = testGroup "Tests"
+  [ compilerTests
+  , evalTests
+  ]
+
+compilerTests :: TestTree
+compilerTests = testGroup "Compiler Tests"
+  [ testCase "Compile single number" $
+      assertEqual "" [PushNum 5] (compile (Num 5 :: Expr String))
```
Comment (kajigor) [inline monads/test/Spec.hs]: use `@?=` if you're not going to pass a message to `assertEqual`.

File: monads/src/HW/Eval.hs [hunk 2]
```diff
@@ -29,8 +29,45 @@ initialState = MachineState [] M.empty
...
+  state <- get
+  put state { getStack = number : getStack state }
+  return (Right ())
+
+execInstr (PushVar variable) = do
+  state <- get
+  case M.lookup variable (getEnv state) of
+    Just value -> do
+      put state { getStack = value : getStack state }
+      return (Right ())
```
Comment (kajigor) [inline monads/src/HW/Eval.hs]: fmap `Right` instead of using `return`

**Review by kajigor — CHANGES_REQUESTED** [review]

File: monads/test/Spec.hs (hunk 1 above)
Comment (npanuhin) [reply monads/test/Spec.hs]: [Fixed](https://github.com/kajigor/fp-2024-cub-fall/pull/96/commits/59ba4ea732e81290ab4a7f0a86e7a19c4d0321d4#diff-0e851a1a04a54af327b8044f27cdd1c408a2fee485bb48c4638e5656bdb5e8a3)

File: monads/src/HW/Eval.hs (hunk 2 above)
Comment (npanuhin) [reply monads/src/HW/Eval.hs]: [Fixed](https://github.com/kajigor/fp-2024-cub-fall/pull/96/commits/59ba4ea732e81290ab4a7f0a86e7a19c4d0321d4#diff-e087aa917692a23becb655a200a54369c94c8557339cef45607c416ec4812624)

**Comment by kajigor** [comment]
Ok.

**Pull request closed without merging**